tests/visidentifier/vismessage_test.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024s EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "visidentifier/vismessage.hpp"
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+class VISMessageTest : public testing::Test { };
+
+/***********************************************************************************************************************
+ * Tests
+ **********************************************************************************************************************/
+
+TEST_F(VISMessageTest, ConstructFromJson)
+{
+    Poco::JSON::Object json;
+    json.set(cActionTagName, "get");
+    json.set(cPathTagName, "test-path");
+    json.set(cRequestIdTagName, "test-request-id");
+
+    std::ostringstream jsonStream;
+    Poco::JSON::Stringifier::stringify(json, jsonStream);
+
+    VISMessage message(jsonStream.str());
+
+    EXPECT_EQ(json.size(), message.GetJSON().size());
+    EXPECT_EQ(message.GetValue<std::string>(cActionTagName), "get");
+    EXPECT_EQ(message.GetValue<std::string>(cPathTagName), "test-path");
+    EXPECT_EQ(message.GetValue<std::string>(cRequestIdTagName), "test-request-id");
+}
+
+TEST_F(VISMessageTest, IsSucceeds)
+{
+    const VISMessage message(VISActionEnum::eGet);
+
+    EXPECT_TRUE(message.Is(VISActionEnum::eGet));
+}
+
+TEST_F(VISMessageTest, IsFails)
+{
+    const VISMessage message(VISActionEnum::eGet);
+
+    EXPECT_FALSE(message.Is(VISActionEnum::eSubscribe));
+}
+
+TEST_F(VISMessageTest, ConstructorSetsAction)
+{
+    const VISMessage message(VISActionEnum::eGet);
+
+    std::string actionTagValue;
+
+    ASSERT_NO_THROW(actionTagValue = message.GetValue<std::string>(cActionTagName));
+    EXPECT_EQ(actionTagValue, "get");
+}
+
+TEST_F(VISMessageTest, SetKeyValue)
+{
+    VISMessage message(VISActionEnum::eGet);
+
+    message.SetKeyValue("key", "value");
+    EXPECT_EQ(message.GetValue<std::string>("key"), "value");
+
+    message.SetKeyValue("key", "value1");
+    EXPECT_EQ(message.GetValue<std::string>("key"), "value1");
+
+    message.SetKeyValue("key", 10);
+    EXPECT_EQ(message.GetValue<int>("key"), 10);
+}
+
+TEST_F(VISMessageTest, GetValueThrowsOnKeyNotFound)
+{
+    VISMessage message(VISActionEnum::eGet);
+
+    try {
+        message.GetValue<std::string>("key-not-found");
```

Comment:
Did you consider ASSERT/EXPECT_ANY_THROW

Reply:
Refactored to use ASSERT